### ISSUE TITLE
Prisma ignores the `database` config from `docker-compose.yml`

### ISSUE THREAD

**Issue by juliusecker** [prisma/prisma#4050]
If you provide a value for `database` in `docker-compose.yml`, Prisma will ignore it, and use `default@default` instead.

### Reproduce:

docker-compose.yml
```yml
version: '3'
services:
  prisma:
    image: prismagraphql/prisma:1.26
    restart: always
    ports:
    - "4466:4466"
    environment:
      PRISMA_CONFIG: |
        managementApiSecret: my-api-secret
        port: 4466
        databases:
          default:
            connector: mysql
            host: my.mysql.host
            port: 3306
            user: prisma
            database: my-custom-db
            password: my-secret
            migrations: true
```
### Run:
```bash
docker-compose up
prisma deploy
```

### Expected:
* Deploy runs successfully 
* Tables under `my-custom-db`.

### Actual:
* Deploy runs successfully 
* Tables under `default@default`

### Versions:
 - Connector: `MySQL`
 - Prisma Server: 1.26 (docker)
 - `prisma` CLI: `npx prisma`
 - OS: Ubuntu 18.04

**Comment by pantharshit00** [comment]
Hi @juliusecker 

That database property is only used when you are using introspection. We need to create these special DB names to support multiple services and stages. Like backend/dev and backend/prod will have two completely isolated DBs: `backend@dev` and `backend@prod`. The default is `default$default` when none of these are specified. That is why I think is expected behavior.

**Comment by pantharshit00** [comment]
Fixed in #4054
